doc/installation.rst

```diff
@@ -264,6 +265,19 @@ In summary:
 
    This installs all of Spyder's dependencies into the environment along with the stable/packaged version of Spyder, and then removes Spyder itself.
 
+   .. note::
+
+      Following the separation of ``spyder-kernels`` into its own modular package in Spyder 3.3.0, you'll need to have the corresponding version of that package available—``0.x`` for Spyder 3 (``3.x`` branch), and ``1.x`` for Spyder 4 (``master`` branch).
+      The above procedure will install the ``0.x`` version; to test the ``master`` branch (Spyder 4), you'll need to install the corresponding ``1.x`` version of ``spyder-kernels``.
+      This can be done via two methods: installing the correct version via ``conda`` (or ``pip``):
+
+      .. code-block:: bash
+
+         conda install spyder-kernels=1.*
+
+      (and using ``conda install spyder-kernels=0.*`` to switch back to the Spyder 3 version), or by ``clone``-ing the `spyder-kernels git repository`_ to somewhere on your path and checking out the appropriate branch (``0.x`` or ``master``) corresponding to the version of Spyder (3 or 4) you would like to run.
```

Comment:
Two things here:

1. Since you need to use pip to install spyder-kernels >= 1.0, you also need to use it to install 0.x, so you need to change `conda install spyder-kernels=0.*` for `pip install spyder-kernels==0.*`.
2. You don't achieve anything by simply cloning the spyder-kernels repo and checking out the right branch. You also need to run `pip install -e .` to install it in your env (I told you that before).

Reply:
> Since you need to use pip to install spyder-kernels >= 1.0, you also need to use it to install 0.x,

I re-wrote that section to clarify the need to use the same command to switch. Also, in the initial instructions, the whole dependency stack, including spyder-kernels, gets installed via ``conda install spyder``. If ``conda install -c conda-forge spyder-kernels=1`` works in the above block as well as ``pip`` (which it seems to in my limited testing), then it avoids this potential problem.

> You also need to run ``pip install -e``

Yup, back-ported.

> I told you that before

Yes, on my PR *after* this one, spyder-ide/spyder#7418 , that had a downstream version of that text. I hadn't yet back-ported the fixes upstream to here.